Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs

```diff
@@ -36,8 +44,9 @@ public override MovementState DecideNextState(Vector3 velocity, Vector3 external
         if (isDisabled || timingState == TimingState.DONE)
         {
             hitDashEarly = false;
-            if (earlyAttackInput)
+            if (attackAllowed)
```

Comment:
This probably shouldn't be happening like this. AttackAllowed isn't supposed to be treated like transition logic internally.